pkg/controller/clusterstorage/clusterstorage_controller.go

```diff
@@ -183,7 +184,36 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// StorageClass already exists - don't requeue
+	// Check to see if modifications have been made to the StorageClass attributes
+	needUpdate := false
+	comparisonSC := sc.DeepCopy()
+	comparisonSC.ObjectMeta = found.ObjectMeta
+
+	// Define a default ReclaimPolicy for comparison
+	if comparisonSC.ReclaimPolicy == nil {
+		deletePolicy := corev1.PersistentVolumeReclaimDelete
+		comparisonSC.ReclaimPolicy = &deletePolicy
+	}
+
+	if !reflect.DeepEqual(comparisonSC, found) {
+		needUpdate = true
+	}
+
+	// If a change has been detected, update the StorageClass.
+	if needUpdate {
+		reqLogger.Info("StorageClass already exists and needs to be updated", "StorageClass.Name", found.Name)
+		err = r.client.Update(context.TODO(), sc)
```

Comment:
Does `sc` has the default annotation? When user changes a parameter *and* makes the SC as non-default, this `Update` makes it back default.

You should copy the annotation (or no annotation) from `found`.

Reply:
Changed this call `Update` on `comparisonSC`, which copies over the ObjectMeta from earlier. Since we're currently using `Update`, instead of `Create`, this seems to apply without issue and keeps the annotations and labels.